library/core/src/intrinsics.rs

```diff
@@ -2210,8 +2210,8 @@ macro_rules! assert_unsafe_precondition {
             #[inline(always)]
             fn runtime$(<$($tt)*>)?($($i:$ty),*) {
                 if !$e {
-                    // abort instead of panicking to reduce impact on code size
-                    ::core::intrinsics::abort();
+                    // don't unwind to reduce impact on code size
+                    ::core::panicking::panic_fmt_nounwind(format_args!("unsafe precondition violated"));
```

Comment:
If you give a moose a muffin...

Can we stringify the predicate and pass that instead?

Reply:
I think I'd be in favor of letting the macro user give a string. But that's for a future PR, first this one needs to work.